From a1fff28a407c2f1dfa72b52753883df4a7b028cc Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Tue, 29 Aug 2017 17:40:33 +0800
Subject: [PATCH 2494/5242] MLK-15348-02 arm: dts: imx7ulp: add focaltech
 touch panel ft5246 support

commit  69bffaed6fd90cf1b1e14531506968d581d9e742 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add focaltech new touch panel ft5246 support.
Set the ft5426 as default panel for dts. If want to use the old panel, then
it needs to boot with imx7ulp-evk-ft5416.dtb file.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
(cherry picked from commit:963fea909ef5e42294cb2e656e5e3870a2171c01)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile               |    1 +
 arch/arm/boot/dts/imx7ulp-evk-ft5416.dts |   17 +++++++++++++++++
 arch/arm/boot/dts/imx7ulp-evk.dts        |    3 ++-
 include/dt-bindings/input/input.h        |    3 +++
 4 files changed, 23 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/imx7ulp-evk-ft5416.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 82cbef5..bbeaf63 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -703,6 +703,7 @@ dtb-$(CONFIG_SOC_IMX7ULP) += \
 	imx7ulp-evk.dtb \
 	imx7ulp-evk-emmc.dtb \
 	imx7ulp-evk-emmc-qspi.dtb \
+	imx7ulp-evk-ft5416.dtb \
 	imx7ulp-evk-sd1.dtb \
 	imx7ulp-evk-lpuart.dtb \
 	imx7ulp-evk-qspi.dtb \
diff --git a/arch/arm/boot/dts/imx7ulp-evk-ft5416.dts b/arch/arm/boot/dts/imx7ulp-evk-ft5416.dts
new file mode 100644
index 0000000..2d86e33
--- /dev/null
+++ b/arch/arm/boot/dts/imx7ulp-evk-ft5416.dts
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2017 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx7ulp-evk.dts"
+
+&lpi2c7 {
+	focaltech@38 {
+		focaltech,panel-type = <FT5416>;
+		focaltech,swap-xy;
+		/delete-property/focaltech,scaling-down-half;
+	};
+};
diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 3bb9df6..c80a117 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -388,6 +388,7 @@
 		reg = <0x38>;
 		interrupt-parent = <&gpio3>;
 		interrupts = <0 0x02>;
+		focaltech,panel-type = <FT5426>;
 		focaltech,reset-gpio = <&gpio3 1 0x01>;
 		focaltech,irq-gpio = <&gpio3 0 0x02>;
 		focaltech,max-touch-number = <5>;
@@ -398,7 +399,7 @@
 		focaltech,keys = <139 102 158>;
 		focaltech,key-y-coord = <2000>;
 		focaltech,key-x-coords = <200 600 800>;
-		focaltech,swap-xy;
+		focaltech,scaling-down-half;
 	};
 };
 
diff --git a/include/dt-bindings/input/input.h b/include/dt-bindings/input/input.h
index bcf0ae1..daa4186 100644
--- a/include/dt-bindings/input/input.h
+++ b/include/dt-bindings/input/input.h
@@ -15,4 +15,7 @@
 #define MATRIX_KEY(row, col, code)	\
 	((((row) & 0xFF) << 24) | (((col) & 0xFF) << 16) | ((code) & 0xFFFF))
 
+#define FT5416     0x54160002
+#define FT5426     0x54260002
+
 #endif /* _DT_BINDINGS_INPUT_INPUT_H */
-- 
1.7.9.5

